mmyolo/models/dense_heads/rtmdet_head.py

```diff
@@ -418,286 +305,130 @@ def loss_by_feat(
         featmap_sizes = [featmap.size()[-2:] for featmap in cls_scores]
         assert len(featmap_sizes) == self.prior_generator.num_levels
 
+        gt_info = self.gt_instances_preprocess(batch_gt_instances, num_imgs)
+        gt_labels = gt_info[:, :, :1]
+        gt_bboxes = gt_info[:, :, 1:]  # xyxy
+        pad_bbox_flag = (gt_bboxes.sum(-1, keepdim=True) > 0).float()
+
         device = cls_scores[0].device
-        anchor_list, valid_flag_list = self.get_anchors(
-            featmap_sizes, batch_img_metas, device=device)
+
+        # If the shape does not equal, generate new one
+        if featmap_sizes != self.featmap_sizes:
+            self.featmap_sizes = featmap_sizes
+            self.multi_level_anchors = self.prior_generator.grid_priors(
```

Comment:
```python
self.featmap_sizes = featmap_sizes
mlvl_priors = self.prior_generator.grid_priors(
                featmap_sizes,
                dtype=cls_scores[0].dtype,
                device=cls_scores[0].device,
                with_stride=True)
self.flatten_priors = torch.cat(mlvl_priors, dim=0)
self.mlvl_priors = [mlvl[:, :2] for mlvl in mlvl_priors]
```
It needs to be replaced with the above writing method, otherwise there will be potential bugs. See https://github.com/open-mmlab/mmyolo/issues/373 for details